suite-native/device/src/hooks/useDeviceConnectionRouteBlacklist.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useFocusEffect, useNavigationState } from '@react-navigation/native';
+
+import { selectIsDeviceRemembered } from '@suite-common/wallet-core';
+import { RootStackRoutes } from '@suite-native/navigation';
+
+import {
+    startDeviceConnectionListening,
+    stopDeviceConnectionListening,
+} from '../middlewares/deviceConnectionNavigationMiddleware';
+
+const DEVICE_CONNECTION_BLACKLISTED_ROUTES: RootStackRoutes[] = [
+    RootStackRoutes.DeviceCompromisedModal,
+    RootStackRoutes.DeviceOnboardingStack,
+    RootStackRoutes.OnboardingStack,
+];
+
+const REMEMBERED_DEVICE_BLACKLISTED_ROUTES: RootStackRoutes[] = [RootStackRoutes.SendStack];
+
+export const useDisableDeviceConnectionOnRouteBlacklist = () => {
```

Comment:
Would it be possible to create a hook to disable/enable the middleware and then use it on relevant screens? What is the benefit of having this global?